internal/auth0/resourceserver/resource_scopes.go

```diff
@@ -0,0 +1,228 @@
+package resourceserver
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/auth0/go-auth0/management"
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/auth0/terraform-provider-auth0/internal/config"
+	"github.com/auth0/terraform-provider-auth0/internal/value"
+)
+
+// NewScopesResource will return a new auth0_resource_server_scopes (1:many) resource.
+func NewScopesResource() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"resource_server_identifier": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Identifier of the resource server that the scopes (permission) are associated with.",
+			},
+			"scopes": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
```

Comment:
In all the other resources we use name for the permission instead of value, so aligning this to name just makes things better IMO. It is a good thing you raised this tho as we should probably rename the soon to be read only value of the resource_server.scopes[0].value to name as well so it's consistent across everywhere.

Reply:
This confusion was also raised in https://github.com/auth0/terraform-provider-auth0/issues/520.

Reply:
So tldr, we should actually change resource_server.scopes.*.value to resource_server.scopes.*.name instead.